### ISSUE TITLE
dev30 causing various parse errors (add/remove effects and variables being interpreted as expressions)

### ISSUE THREAD

**Issue by ExtraCreep** [bensku/Skript#798]
82 errors for good syntax. What ?
https://pastebin.com/U0vHAHvG

Skript-dev30d (latest)
Umbaska 3.3 (latest)
SkQuery 3.5.9-Lime (latest)
TusKe 1.8.1 (latest)
Skellett 1.9.6 (latest)
ProtocolLib latest

Spigot 1.11.2

> [21:20:00 INFO]: This server is running CraftBukkit version git-Spigot-3fb9445-6e3cec8 (MC: 1.11.2) (Implementing API version 1.11.2-R0.1-SNAPSHOT)

**Comment by TheBentoBox** [comment]
You'll note the dev30 release post has a billion warnings about instabilities, such as:
```
First, a warning: this is not stable build, at least not until way more testing has been done. Test it before putting it to your public server, especially if you use addons.
```
and
```
Skript now has support for parsing scripts asynchronously. It is somewhat unstable, so it should be used only for testing in non-public servers
```
so you could do with a less dramatic title.

As for these issues, you may do better reporting these on #779 to keep all of dev30's issue more centralized, but I'll leave it up to bensku if he wants to leave this open as its own separate issue; I'm not sure if these `add` and `remove` effect errors are already reported or not.

Oh, and while you're correct that most of these don't seem to be your fault, 4 of your errors are because you're using `send` with no target in a function event, which is **not** "good syntax" as it's not valid. You need to specify who's receiving the message in a function event, be it the console or a player who was passed in as an argument.

**Comment by ExtraCreep** [comment]
Hi. 

> you're using send with no target in a function event
With another version of Skript (not sure if it was a version made by bensku), I could use "send" without specifying anyone (if I don't specifying, it was like "message"). Sorry for this error.

> You'll note the dev30 release post has a billion warnings about instabilities, such as:

I know it, but with a stable version of bensku's Skript, it was the same thing.

**Comment by TheBentoBox** [comment]
If you're getting these errors even when using (for example) dev25 it's quite possibly due to an addon.

**Comment by ExtraCreep** [comment]
I go to test with removing every addon.

**Comment by ExtraCreep** [comment]
And for the "send" code: I set the target.

`send "blabla" to {_e}`, `send "blabla" to {_p}` etc

**Comment by ExtraCreep** [comment]
@TheBentoBox No changes. Tried without add-ons and with different versions of Skript.

**Comment by bensku** [comment]
Cannot reproduce with following snippet:
`on script load:
  set {_e} to console
  send "blabla" to {_e}`

Please write a small test script that encounters your issue and post it here. I don't see any issue here, but perhaps I just missed something obvious.

I did not use asynchronous parsing for my tests, did you use it?

**Comment by TheBentoBox** [comment]
I'm currently using my normal setup (many addons, synchronous Skript loading) on 1.12.1 with dev30d and am still getting tons of errors. A bunch of them are related to referencing the loop-value of an entity loop, such as here:

```
[13:23:00 ERROR]: [Skript] There's no loop that matches 'loop-player' (ValorBosses.sk, line 102: set {_p} to loop-player')
[13:23:00 ERROR]: [Skript] There's no loop that matches 'loop-player' (ValorBosses.sk, line 103: send "&a[MobArena]&f The spire defense mechanisms activate &ePositron Cannon&f!" to loop-player')
```

I'm also still getting the "registerCaller" error with functions, [pastebin](https://pastebin.com/nh2Me6Lc), and am getting some `not a valid item data` when referencing lists:

```
[13:23:12 ERROR]: [Skript] '::*} < 2' is not a valid item data (zTabCompletions.sk, line 17: if size of {_split::*} < 2:')
[13:23:11 ERROR]: [Skript] '::%offline player-argument's uuid%::*} > 0' is not a valid item data (ValorShop.sk, line 463: if size of {valorLog::%offline player-argument's uuid%::*} > 0:')
```

And `List is missing 'and' or 'or'` warnings with a default effect:
```
[13:22:35 WARN]: [Skript] List is missing 'and' or 'or', defaulting to 'and': {_rgb::1}, {_rgb::2}, {_rgb::3} (DyeSmear.sk, line 102: dye player's tool ({_rgb::1}, {_rgb::2}, {_rgb::3})')
```

Given that I'm using a number of addons, I expect one of them might be causing issues due to it not being properly updated to 1.12. I'll look more into that soon.

**Comment by bensku** [comment]
The function error thing has been fixed. And actually, that might fix other problems too. Skript rarely fails gracefully; instead exceptions literally fall down the stack and break everything in their way. Sometimes this causes only small issues, but...
